src/Hocon.Configuration.Test/ConfigurationSpec.cs

```diff
@@ -427,5 +431,49 @@ public void ShouldSerializeFallbackValues()
             c.ToString().Should().NotContain("other-key", "Fallback values are ignored by default");
             c.ToString(true).Should().Contain("other-key", "Fallback values should be displayed when requested");
         }
+
+        /// <summary>
+        /// Source issue: https://github.com/akkadotnet/HOCON/issues/175
+        /// </summary>
+        [Fact]
+        public void ShouldDeserializeFromJson()
+        {
+            var settings = new JsonSerializerSettings
+            {
+                PreserveReferencesHandling = PreserveReferencesHandling.Objects,
+                Converters = new List<JsonConverter>(),
+                NullValueHandling = NullValueHandling.Ignore,
+                DefaultValueHandling = DefaultValueHandling.Ignore,
+                MissingMemberHandling = MissingMemberHandling.Ignore,
+                ObjectCreationHandling = ObjectCreationHandling.Replace,
+                ConstructorHandling = ConstructorHandling.AllowNonPublicDefaultConstructor,
+                TypeNameHandling = TypeNameHandling.All,
+                ContractResolver = new AkkaContractResolver()
+            };
+            
+            var json = "{\"$id\":\"1\",\"$type\":\"Hocon.Config, Hocon.Configuration\",\"Root\":{\"$type\":\"Hocon.HoconEmptyValue, Hocon\",\"$values\":[]},\"Value\":{\"$type\":\"Hocon.HoconEmptyValue, Hocon\",\"$values\":[]},\"Substitutions\":{\"$type\":\"Hocon.HoconSubstitution[], Hocon\",\"$values\":[]},\"IsEmpty\":true}";
+            var restored = JsonConvert.DeserializeObject<Config>(json, settings);
+            restored.IsEmpty.Should().BeTrue();
+        }
+        
+        internal class AkkaContractResolver : DefaultContractResolver
```

Comment:
Identical to what we have inside the Akka.NET project, in case that's now clear to other project members:

https://github.com/akkadotnet/akka.net/blob/0a8c0390ee24927e3c381a99bca079ebc7e5b8e9/src/core/Akka/Serialization/NewtonSoftJsonSerializer.cs#L187-L205